shared/actions/chat/inbox.js

```diff
@@ -731,35 +735,26 @@ function* _inboxSynced(action: ChatGen.InboxSyncedPayload): Saga.SagaGenerator<a
   const conversation = Constants.getSelectedConversationStates(state)
   if (conversation) {
     const inbox = Constants.getInbox(state, selectedConversation)
+    const lastMessageIDWeAreShowing = Constants.lastMessageID(state, selectedConversation)
+
+    // Check the data we are given first, then our state
+    const knownMaxMessageID: ?number = latestMaxMsgID[selectedConversation] || (inbox ? inbox.maxMsgID : null)
 
-    const messageKeys = Constants.getConversationMessages(state, selectedConversation)
-    const lastMessageKey = messageKeys.last()
     let numberOverride
-    if (lastMessageKey) {
-      const lastMessage = Constants.getMessageFromMessageKey(state, lastMessageKey)
+    if (lastMessageIDWeAreShowing && knownMaxMessageID) {
+      const lastRpcMessageIdWeAreShowing = Constants.messageIDToRpcMessageID(lastMessageIDWeAreShowing)
+
       // Check to see if we could possibly be asking for too many messages
-      if (
-        lastMessage &&
-        typeof lastMessage.rawMessageID === 'number' &&
-        lastMessage.rawMessageID &&
-        inbox &&
-        inbox.maxMsgID
-      ) {
-        numberOverride = inbox.maxMsgID - lastMessage.rawMessageID
-
-        if (numberOverride > tooManyMessagesToJustAppendOnStale) {
-          console.log(
-            'Doing a full load due to too many old messages',
-            inbox.maxMsgID - lastMessage.rawMessageID
-          )
-          yield Saga.all([
-            Saga.put(ChatGen.createClearMessages({conversationIDKey: selectedConversation})),
-            yield Saga.put(
-              ChatGen.createLoadMoreMessages({conversationIDKey: selectedConversation, onlyIfUnloaded: false})
-            ),
-          ])
-          return
-        }
+      numberOverride = knownMaxMessageID - lastRpcMessageIdWeAreShowing
+      if (numberOverride > tooManyMessagesToJustAppendOnStale) {
+        console.log('Doing a full load due to too many old messages', numberOverride)
+        yield Saga.all([
+          Saga.put(ChatGen.createClearMessages({conversationIDKey: selectedConversation})),
+          yield Saga.put(
```

Comment:
this was in the old code, but should this be yielding inside a `Saga.all`?

Reply:
ah, good catch!